Minio/MinioClient.cs

```diff
@@ -594,7 +613,18 @@ private void LogRequest(IRestRequest request, IRestResponse response, double dur
 
             this.logger.LogRequest(requestToLog, responseToLog, durationMs);
         }
+
```

Comment:
```suggestion
            if (response.StatusCode.Equals(HttpStatusCode.NotFound)
                && response.Request.Method.Equals(Method.GET) && errResponse.Code == "NoSuchBucket")
            {
                    throw new BucketNotFoundException(errResponse.BucketName, "Not found.");

            }
```
Add this just before throwing [MinioException](https://github.com/minio/minio-dotnet/blob/master/Minio/MinioClient.cs#L505) in ParseError() @ig-sinicyn